### DISCUSSION TITLE
Heatmap based on a cluster of samples, not between clusters

### DISCUSSION THREAD

**Discussion by strawberry789** [GreenleafLab/ArchR#1201] [Questions / Documentation]
How is it possible to plot a heatmap that's based on a particular cluster of different samples rather than different clusters of the integrated dataset?

**Comment by rcorces** [comment]
Sorry - I do not think I understand the question.

**Reply by strawberry789** [reply]
Sorry, I should phrase it better.  Instead of a heatmap that compares between clusters of an integrated dataset, is it possible to generate a heatmap that compares a specific cluster between the different samples in the integrated dataset? - Thanks.

**Reply by rcorces** [reply]
I'm not sure if this answers your question but comparisons in ArchR are generally driven by selecting a column in `cellColData`. So you should think about adding a new column in `cellColData` that represents the comparison that you want to make.

**Reply by strawberry789** [reply]
The `cellColData` would be `sample`.  Would something like the following command work:

`heatmapEM <- plotEnrichHeatmap(enrichMotifs, n = 20, cutOff = 10, transpose = TRUE, colorBy = "cellColData", name = "Sample")`

Instead of `colorBy`, what should the option name be?

**Reply by strawberry789** [reply]
I was able to generate a heatmap that consists of only the cluster of cells I wanted using this function:

`Distalheatmap <- plotEnrichHeatmap(
  seEnrich = enrichMotifs[,1],
  pal = paletteContinuous(set = "comet", n = 100),
  n = 10,
  cutOff = 20,
  pMax = Inf,
  clusterCols = TRUE,
  binaryClusterRows = TRUE,
  labelRows = TRUE,
  rastr = TRUE,
  transpose = FALSE,
  returnMatrix = FALSE,
  logFile = createLogFile("plotEnrichHeatmap")
)`

but is there a `groupBy` parameter such that I could separate the samples in this integrated dataset?

Thanks!
